department-of-veterans-affairs/caseflow
# MattT/CASEFLOW-4600 Add Assigned Tab to EMO Queue

Resolves [CASEFLOW-4600](https://vajira.max.gov/browse/CASEFLOW-4600)


### Description
Adding an 'Assigned' tab to Caseflow Queue for the EMO organization.


### Acceptance Criteria
- [ ] Determination of the behavior and triggers of the Assigned tab
- [ ] When cases are assigned to one of the three (3) RPO organizations, they appear on the Assigned tab
- [ ] If the RPO organization sends it to board intake, the appeal remains in EMO’s Assigned tab (it is also moved to the Completed tab of the RPO queue)
- [ ] If the RPO organization sends the appeal back to the EMO office, it is removed from the Assigned tab goes to the Unassigned tab
- [ ] If the EMO organization sends an appeal to BVA Intake, regardless of the reason selected (not an education issue or appeal ready for review), the appeal shows on the EMO's "Assigned" tab.

### Testing Plan
1. Ensure that you have the `edu_predocket_appeals` feature toggle enabled. You can paste `FeatureToggle.enable!(:edu_predocket_appeals)` into the Rails console in order to enable it.
2. If you have not yet reseeded your database with the seed file updates from the feature branch, run `make seed-dbs`. The updated seed files are present in this branch.
3. Open Caseflow and switch to an Intake user.
4. Navigate to Caseflow Intake and begin the 10182 entry flow.
5. Whenever you arrive at the Add/Remove issues page, add an issue with the benefit type of 'Education' and select 'Yes' in the pre-docketing radio field.
6. Submit the appeal.
7. Switch to an EMO user.
8. Navigate to the EMO Organization's Queue. It should be [at this address](http://localhost:3000/organizations/edu-emo?tab=education_assigned&page=1).
9. Ensure that the appeal is **not present**. Whenever an appeal is first assigned to the EMO it will be present in the 'Unassigned' tab until an action is taken upon it.
10. Open up your Rails console.
11. Run `a = Appeal.order("created_at").last` to get the appeal you've just submitted.
12. Due to the lack of full UI functionality at this time, you must manually edit the status of the appeal's task(s) in order to get it to appear in the Assigned tab. I encourage you to ensure that the appeal appears in the tab with all three statuses:

```
a.tasks.last.update!(status: Constants.TASK_STATUSES.on_hold)
a.tasks.last.update!(status: Constants.TASK_STATUSES.completed)
a.tasks.last.update!(status: Constants.TASK_STATUSES.cancelled)
```
13. To simulate a scenario where the EMO has sent an appeal to an RPO, enter the following commands. Ensure that the task remains in the EMO's assigned tab afterwards.
```
a.tasks.last.update!(status: Constants.TASK_STATUSES.on_hold)

EducationAssessDocumentationTask.create!(parent: a.tasks.last, appeal: a, assigned_at: Time.zone.now,assigned_to: EduRegionalProcessingOffice.first)
```

14. Set the `EducationDocumentSearchTask` to `completed`.

`a.tasks.last.update!(status: Constants.TASK_STATUSES.completed)`

15. If you simulated adding an RPO task, you can set that to be completed as well:

`a.tasks.last.children.first.update!(status: Constants.TASK_STATUSES.completed)`

16. Set the parent, `PreDocketTask` (BVA Intake), to a status of completed. Ensure that the appeal is no longer present in the Assigned tab afterwards (it will move to the completed tab). This simulates BVA Intake sending a pre-docket appeal to docket.

`a.tasks.last.parent.first.update!(status: Constants.TASKS_STATUSES.completed)`

17. Now set the  `PreDocketTask` to a status of `cancelled`. This simulates BVA Intake removing all of the issues from an appeal to cancel it. The appeal should not appear in the Assigned tab once you have done so.
18. To simulate an RPO sending an appeal back to EMO, you can run the following commands. The appeal should no longer appear in the Assigned tab (it would be in the in the unassigned tab).
```
# If you have adjusted the status of the Pre Docket Task, set it back to on hold
a.tasks.last.parent.first.update!(status: Constants.TASKS_STATUSES.on_hold)

# Set the RPO task to completed if it isn't already.
a.tasks.last.children.first.update!(status: Constants.TASK_STATUSES.completed)

# Set EMO task to status: assigned, which is the status it will have if an RPO sends an appeal to EMO.
a.tasks.last.update!(status: Constants.TASK_STATUSES.assigned)
```


### User Facing Changes

 There is no before image available for this change as it is a brand new UI addition.

![Edu EMO Assigned Tab](https://user-images.githubusercontent.com/99351305/163846138-aa0c5da1-7005-4925-8b00-8c27bc79320b.png)


### Code Documentation Updates
